YashGoyal28/main.js

```diff
@@ -0,0 +1,314 @@
+    else virturn[1]=0; 
+    var move = getbestmove();
+    if(move == "draw"){
+        move = onlyonemove;
+    }
+    document.getElementById(move+1).innerHTML = "<span>" + marker[turn] + "</span>";
+    board[move]=marker[turn];
+    switch(checkwin(move+1,turn)){
+        case "win":
+                document.getElementById("playerturn").innerText = Players[turn] + " Wins";
+                document.getElementById("heading").innerHTML = "-> New Game <-";
+                break;
+        case "draw":
+                document.getElementById("playerturn").innerText = " Draw! ";
+                document.getElementById("heading").innerHTML = "-> New Game <-";
+                break;
+        case "no":
+                changeturn();
+                break;
+    }
+}
+
+function getbestmove(){
+    var icandraw = -1;
+    for(var i=0 ; i<9 ; i++){
+        if(board[i] == "#"){
+            board[i] = marker[virturn[0]]; 
+            switch(checkwin(i+1, virturn[0])){
+                case "win":
+                    board[i] = "#";
+                    return i;
+                case "draw":
+                    board[i] = "#";
+                    icandraw = i;
+                    onlyonemove = icandraw;
+                    return "draw";
+                case "no":
+                        switch(virplayer()){
+                            case "ulose":
+                                board[i] = "#";
+                                break;
+                            case "uwin":
+                                board[i] = "#";
+                                return i;
+                            case "draw":
+                                board[i]="#";
+                                icandraw = i;
+                                break;
+                        }
+            }
+        }
+    }
+    if(icandraw != -1){ 
+        onlyonemove = icandraw;
+        return "draw";}
+    else return "ilose";
+}
+
+function virplayer(){
+    var icandraw = -1;
+    for(var i=0; i<9; i++){
+        if(board[i] == "#"){
+            board[i] = marker[virturn[1]];  
+            switch(checkwin(i+1, virturn[1])){
+                case "win":
+                    board[i] = "#";
+                    return "ulose";
+                case "draw":
+                    board[i] = "#";
+                    return "draw";
+                case "no":
+                    switch(getbestmove()){
+                        case "ilose":
+                            board[i] = "#";
+                            return "ulose";
+                        case "draw":
+                            board[i] = "#";
+                            icandraw = i;
+                            break;
+                        default:
+                            board[i] = "#";
+                    }
+            }
+        }
+    }
+    if(icandraw != -1) return "draw";
+    else return "uwin";
+}
+
+function mode1(){
+    location.href="index_2.html";
```

Comment:
Name your files in all lowercase letters. The index_2.html here redirects to error page because you have named the file Index_2.html instead.